test/evp_extra_test2.c

```diff
@@ -414,6 +414,41 @@ static int test_d2i_PrivateKey_ex(int testid)
     return ok;
 }
 
+static int test_PEM_read_bio_negative(int testid)
+{
+    int ok = 0;
+    OSSL_PROVIDER *provider = NULL;
+    BIO *key_bio = NULL;
+    EVP_PKEY *pkey = NULL;
+
+    if (!TEST_ptr(key_bio = BIO_new_mem_buf(keydata[testid].kder, keydata[testid].size)))
+        goto err;
+    ERR_clear_error();
+    if (!TEST_ptr_null(pkey = PEM_read_bio_PrivateKey(key_bio, NULL, NULL, NULL)))
+        goto err;
+    if (!TEST_int_ne(ERR_peek_error(), 0))
+        goto err;
+    if (!TEST_ptr(provider = OSSL_PROVIDER_load(NULL, "default")))
```

Comment:
Should you be using a libctx here - just in case this affects other tests that could get added later.

Reply:
The provider is unloaded later. The NULL libctx is supposed to have only the null provider loaded so loading/unloading the default provider should not affect the other tests.